### PR TITLE
feat: Enable HIP-904 transactions and fix when empty transfers are submitted in `TokenAirdrop` transaction

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes https://github.com/hashgraph/hedera-services/issues/15200

- Enables TokenAirdrop transactions on develop, that need to be enabled in 0.54 by default
- Fixes a `FAIL_INVALID` transaction hen token transfer list is empty

### CODE REVIEW

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java [hunk 1]
```diff
@@ -60,10 +60,10 @@ public record TokensConfig(
         @ConfigProperty(value = "maxMetadataBytes", defaultValue = "100") @NetworkProperty int tokensMaxMetadataBytes,
         @ConfigProperty(value = "balancesInQueries.enabled", defaultValue = "true") @NetworkProperty
                 boolean balancesInQueriesEnabled,
-        @ConfigProperty(value = "airdrops.enabled", defaultValue = "false") @NetworkProperty boolean airdropsEnabled,
-        @ConfigProperty(value = "airdrops.cancel.enabled", defaultValue = "false") @NetworkProperty
+        @ConfigProperty(value = "airdrops.enabled", defaultValue = "true") @NetworkProperty boolean airdropsEnabled,
+        @ConfigProperty(value = "airdrops.cancel.enabled", defaultValue = "true") @NetworkProperty
                 boolean cancelTokenAirdropEnabled,
-        @ConfigProperty(value = "airdrops.claim.enabled", defaultValue = "false") @NetworkProperty
+        @ConfigProperty(value = "airdrops.claim.enabled", defaultValue = "true") @NetworkProperty
```
Comment (david-bakin-sl) [inline hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java:66]: Are these config changes to be merged to develop in this PR?

**Review by david-bakin-sl — CHANGES_REQUESTED** [review]
Config changes are probably wrong in a "fix" PR?

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java:66]: Those should have already been enabled by default on develop @david-bakin-sl . This feature is going in 0.54. I have changed the title of the PR 🙂

Comment (david-bakin-sl) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java:66]: Changing the title makes this make sense, TY!

**Review by david-bakin-sl — DISMISSED** [review]
LGTM!

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java:66]: Thank you!

**Review by david-bakin-sl — APPROVED** [review]

**Review by david-bakin-sl — DISMISSED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenCancelAirdropTest.java [hunk 2]
```diff
@@ -547,6 +547,14 @@ final Stream<DynamicTest> FTAndNFTToDifferentReceivers() {
                         .hasKnownStatus(INVALID_PENDING_AIRDROP_ID));
     }
 
+    @HapiTest
+    @DisplayName("empty cancel should fail")
+    final Stream<DynamicTest> emptyCancelFails() {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenCancelAirdropTest.java:552]: For the cancel we already have a test for this: `cancelWithAnEmptyAirdropList`

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java [hunk 3]
```diff
@@ -1220,6 +1220,14 @@ final Stream<DynamicTest> claimWithSenderAccount() {
                         .hasKnownStatus(INVALID_SIGNATURE)));
     }
 
+    @HapiTest
+    @DisplayName("empty claim should fail")
+    final Stream<DynamicTest> emptyClaimFails() {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java:1225]: We have the same test for claim as well - `tokenClaimWithNoPendingAirdrop`

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenCancelAirdropTest.java (hunk 2 above)
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenCancelAirdropTest.java]: Deleted

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java (hunk 3 above)
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java]: Deleted

**Review by vtronkov — APPROVED** [review]
LGTM!

**Pull request merged**